### PR TITLE
Fix dead assignment detected by clang CI

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Comment by ghost** [comment]
Rebase if possible

File: librz/asm/arch/z80/z80asm.c [hunk 1]
```diff
@@ -1507,7 +1507,6 @@ static int assemble(const char *str, unsigned char *_obuf) {
 				eprintf ("else without if\n");
 				break;
 			}
-			noifcount = 1;
```
Comment (ret2libc) [inline librz/asm/arch/z80/z80asm.c]: mmm are you really sure this is dead? why is this dead but not the `noifcount--` in the next `case`?

Comment (ghost) [reply librz/asm/arch/z80/z80asm.c]: I want to open a third PR to refacto this function.
Indeed there is a dead while (line 1535) that is breaking the analysis.

File: librz/bin/format/pe/pe.c [hunk 2]
```diff
@@ -1443,16 +1443,11 @@ static int bin_pe_init_imports(struct PE_(rz_bin_pe_obj_t) * bin) {
 	int indx = 0;
 	int rr, count = 0;
 	int import_dir_size = data_dir_import->Size;
-	int delay_import_dir_size = data_dir_delay_import->Size;
 	/// HACK to modify import size because of begin 0.. this may report wrong info con corkami tests
 	if (!import_dir_size) {
 		// asume 1 entry for each
 		import_dir_size = data_dir_import->Size = 0xffff;
 	}
-	if (!delay_import_dir_size) {
```
Comment (ret2libc) [inline librz/bin/format/pe/pe.c:1452]: I think this shouuld not completely be removed but it should become something like:
```C
if (!data_dir_delay_import->Size) {
  data_dir_delay_import->Size = 0xffff;
}
```
Unless also `data_dir_delay_import->Size` is dead...

File: librz/asm/arch/z80/z80asm.c (hunk 1 above)
Comment (ret2libc) [reply librz/asm/arch/z80/z80asm.c]: Mmm if so, could you remove this deletion? Just to be sure... I'm no 100% convinced of all this piece of code :/

**Review by ret2libc — APPROVED** [review]

**Pull request merged**